We can virtually taste the reality. (VR Support)

Taeir opened the pull request:
With the addition of VR Support, the tests and the TestUtil have had to be changed as well.

The NoVRMouseManager is not tested, and as such only has 20% coverage. This is the case, because it is only used for testing, and will not be present in the final game. It makes it so that we can look around with the mouse or controller, when no VR device is attached.

This closes #123

Taeir committed:
Implemented VR Code and fixed tests with VR

Currently, we can no longer look around without oculus.

Taeir committed:
Removed some unused code

Taeir committed:
VR Fixing in progress

Taeir committed:
Merge branch 'fix-controller' into feature-vr

Conflicts:
	src/main/java/nl/tudelft/contextproject/Main.java

Taeir committed:
Improved Controls with VR + Fixes

Fixed failing tests

Taeir committed:
Merge branch 'dev-sprint5' into feature-vr

Conflicts:
	src/main/java/nl/tudelft/contextproject/Main.java
	src/main/java/nl/tudelft/contextproject/controller/Controller.java
	src/test/java/nl/tudelft/contextproject/model/entities/DrawableTest.java

Taeir committed:
Fixed problems with TestUtil mockGame

Taeir committed:
Merge branch 'dev-sprint5' into feature-vr

Conflicts:
	src/main/java/nl/tudelft/contextproject/Main.java
	src/main/java/nl/tudelft/contextproject/controller/GameController.java
	src/test/java/nl/tudelft/contextproject/model/entities/DrawableTest.java

Geweldig commented on src/main/java/nl/tudelft/contextproject/Main.java:
```diff
@@ -91,10 +121,13 @@ public static void setInstance(Main main) {
 			controller = c;
-			stateManager.attach(controller);
+			
+			if (controller != null) {
```
Doesn't this if check the same thing as line 122 as a result of line 126?

ChielBruin replied:
On line 123 the old controller is checked, on 128 the new controller (set on line 126)

Geweldig replied:
on line 122 c is checked to not be null, on line 126 the controller is set to c, on line 128 the controller is checked (which has just been set to c, which has already been checked)

Taeir replied:
No, @ChielBruin is correct.
The first line checks if `c` is not equal to the old controller
The second if checks if the old controller is non null.
Then `controller = c` makes it so that `controller` points to the new controller.
Then the last check is for the `controller`, which is the new controller.

Geweldig replied:
Looks like I either forgot how to read or my git diff was broken, as I swear line 122 said c != null

ChielBruin commented on src/main/java/nl/tudelft/contextproject/Main.java:
```diff
@@ -147,31 +180,21 @@ public void setGuiNode(Node gn) {
 	@Override
 	public void simpleInitApp() {
-		setDisplayFps(debugHud);
-		setDisplayStatView(debugHud);
```
Because this is removed, the option to show it should also be removed.

Taeir replied:
_Resolved in 750ce96_

ChielBruin commented on src/main/java/nl/tudelft/contextproject/Main.java:
```diff
@@ -229,10 +256,11 @@ protected void setupControlMappings() {
+			im.addMapping("Pickup", new KeyTrigger(KeyInput.KEY_E));
 		}
 
+		im.addMapping("Exit", new KeyTrigger(KeyInput.KEY_ESCAPE));
```
You mapped KEY_ESCAPE to "exit", but the Y on the controller to "Unmapped"

Taeir replied:
This is intentional, and for multiple reasons:
- having one of the main controller buttons as Quit game button is inconvenient and confusing. When players try to figure out the controls, they will accidentally quit the game.
- the ESC key will always stop the game
- when mapping the Y on the controller to Exit, the game will exit upon startup (jMonkey...).

ChielBruin commented:
I made some small comments, but the rest looks good!

Taeir committed:
Removed debugHud

The debughud is incompatible with VR

I have changed the tests from testing the debugHud to testing the QR code
instead, since we did not have tests for that yet, and they are virtually
the same.

Taeir committed:
Merge branch 'dev-sprint5' into feature-vr

Conflicts:
	src/main/java/nl/tudelft/contextproject/controller/GameController.java
	src/main/java/nl/tudelft/contextproject/model/entities/VRPlayer.java
	src/test/java/nl/tudelft/contextproject/controller/GameControllerTest.java
	src/test/java/nl/tudelft/contextproject/model/entities/VRPlayerTest.java

Geweldig commented:
Now that all comments have been resolved, this can and should be merged.
